client/js/libs/handlebars/ircmessageparser/parseStyle.js

```diff
@@ -64,20 +94,23 @@ function parseStyle(text) {
 		case COLOR:
 			emitFragment();
 
+			// Go one step further to find the corresponding color
 			colorCodes = text.slice(position + 1).match(colorRx);
 
 			if (colorCodes) {
 				textColor = Number(colorCodes[1]);
-				bgColor = Number(colorCodes[2]);
-				if (Number.isNaN(bgColor)) {
-					bgColor = undefined;
+				if (colorCodes[2]) {
+					bgColor = Number(colorCodes[2]);
 				}
+				// Color code length is > 1, so bump the current position cursor by as
+				// much (and reset the start cursor for the current text block as well)
 				position += colorCodes[0].length;
+				start = position + 1;
```

Comment:
I moved this here from L80 thinking there could be an issue by moving the `start` cursor when no colors are found. In reality, L80 was a no-op (because `position` is not changed there so `start` did not get changed either) but I moved it still as it was useless in L80. Really what this means is: only update `start` if `position` was updated.